CombinatorialGames/Game/Specific/Domineering.lean

```diff
@@ -21,154 +24,133 @@ disjoint parts of the chessboard give sums of games.
 
 namespace PGame
 
-namespace Domineering
-
 open Function
 
-/-- The equivalence `(x, y) ↦ (x, y+1)`. -/
-@[simps!]
-def shiftUp : ℤ × ℤ ≃ ℤ × ℤ :=
-  (Equiv.refl ℤ).prodCongr (Equiv.addRight (1 : ℤ))
+/-- A Domineering board is an arbitrary finite subset of `ℤ × ℤ`. -/
+def Domineering := Finset (ℤ × ℤ) deriving DecidableEq
+@[match_pattern] def toDomineering : Finset (ℤ × ℤ) ≃ Domineering := Equiv.refl _
+@[match_pattern] def ofDomineering : Domineering ≃ Finset (ℤ × ℤ) := Equiv.refl _
 
-/-- The equivalence `(x, y) ↦ (x+1, y)`. -/
-@[simps!]
-def shiftRight : ℤ × ℤ ≃ ℤ × ℤ :=
-  (Equiv.addRight (1 : ℤ)).prodCongr (Equiv.refl ℤ)
+@[simp] theorem toDomineering_ofDomineering (a : Domineering) :
+  toDomineering (ofDomineering a) = a := rfl
+@[simp] theorem ofDomineering_toDomineering (s : Finset (ℤ × ℤ)) :
+  ofDomineering (toDomineering s) = s := rfl
 
-/-- A Domineering board is an arbitrary finite subset of `ℤ × ℤ`. -/
-abbrev Board :=
-  Finset (ℤ × ℤ)
+namespace Domineering
+
+unsafe instance : Repr Domineering := Finset.instRepr
```

Comment:
Right! I forgot that `Finset` order is locked under `unsafe`.

Reply:
We could use `Finset.sort` to give a safe instance. But I don't think that's really necessary - when are we going to want to prove things about `Repr` anyways?